Automattic/wp-calypso
# Use Total Sale Discount From Api for Pricing Intro Banner

#### Changes proposed in this Pull Request

- Add selector for new `final_discount` field on jetpackSaleCoupon ( D74110-code )
- Use the above selector to provide the sale discount for the Jetpack Pricing page
- Add request status to jetpackSaleCoupon state
- Use jetpackSaleCoupon request state to show IntroPricingBanner in a loading state if need be

#### Testing instructions

1. Sandbox `public-api.wordpress.com` and apply D74110-code ( if not already merged )
2. Follow testing instructions in D74110-code to create a test marketing coupon
3. Boot branch as Jetpack Cloud or use the live link and navigate to `/pricing`.
4. Verify that the intro banner has a loading state before the coupon request finishes. <img width="581" alt="Screen Shot 2022-02-01 at 10 22 56" src="https://user-images.githubusercontent.com/2810519/152028434-8ca056ee-497f-4f7d-be57-67a384554151.png">
5. Verify that the `final_discount` from the coupon is used instead of the default 50%. <img width="1282" alt="Screen Shot 2022-02-01 at 10 25 55" src="https://user-images.githubusercontent.com/2810519/152028488-1dfe5597-ba2d-4410-98f8-bc83ad69111a.png">
6. Un-sandbox `public-api.wordpress.com.`
7. Reload the pricing page and verify that it still has a loading state, but it uses the default 50% when loaded.
